relayer/relayer.go

```diff
@@ -43,14 +48,25 @@ func NewRelayerFromConfig(config *config.RelayerConfig, logger sdklogging.Logger
 	}
 
 	return &Relayer{
-		logger:      logger,
 		rpcClient:   rpcClient,
 		rpcUrl:      config.RpcUrl,
 		daAccountId: config.DaAccountId,
 		nearClient:  nearClient,
+		logger:      logger,
+		listener:    &SelectiveListener{},
 	}, nil
 }
 
+func (r *Relayer) WithMetrics(registry *prometheus.Registry) error {
```

Comment:
Most times I've seen this `With` pattern in go it's used as a Builder, so it returns the object itself. Since this is fallible and not the same use-case, we may want to rename this.